Zend/zend_compile.c

```diff
@@ -5243,6 +5249,14 @@ static zend_type zend_compile_typename(zend_ast *ast, zend_bool force_allow_null
 	}
 }
 /* }}} */
+zend_bool zend_is_null_const(zval constant) /* {{{ */
+{
+	return Z_TYPE(constant) == IS_NULL
+			|| (Z_TYPE(constant) == IS_CONSTANT_AST
+				&& Z_ASTVAL(constant)->kind == ZEND_AST_CONSTANT
+				&& strcasecmp(ZSTR_VAL(zend_ast_get_constant_name(Z_ASTVAL(constant))), "NULL") == 0);
```

Comment:
I believe this code is redundant nowadays, and just checking `Z_TYPE(constant) == IS_NULL` is sufficient. You can try that and see if any tests fail (they shouldn't...)